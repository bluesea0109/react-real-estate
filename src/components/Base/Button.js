import { Button } from 'semantic-ui-react';
import styled from 'styled-components';
import * as brandColors from '../utils/brandColors';

export default styled(Button)`
  &&& {
    min-width: 100px;
    margin: 0.25em;
    &&&.button {
      color: ${brandColors.grey05};
      background-color: #eaeaea;
      &:hover {
        background-color: #e0e0e0;
      }
      &.primary {
        background-color: ${brandColors.primary};
        color: white;
        &:hover,
        &:active,
        &:focus {
          background-color: ${brandColors.primaryHover};
          box-shadow: 0 0 0 2px ${brandColors.primaryHover} inset !important;
          color: ${brandColors.primary};
        }
        &.inverted {
          background-color: transparent;
          box-shadow: 0 0 0 2px ${brandColors.primary} inset !important;
          color: ${brandColors.primary};
          &:hover,
          &:active,
          &:focus {
            background-color: ${brandColors.primary};
            color: white;
          }
        }
      }
      &.secondary {
        background-color: ${brandColors.secondary};
        color: white;
        &:hover,
        &:active,
        &:focus {
          background-color: transparent;
          box-shadow: 0 0 0 2px ${brandColors.secondary} inset !important;
          color: ${brandColors.secondary};
        }
        &.inverted {
          background-color: transparent;
          box-shadow: 0 0 0 2px ${brandColors.secondary} inset !important;
          color: ${brandColors.secondary};
          &:hover,
          &:active,
          &:focus {
            background-color: ${brandColors.secondary};
            color: white;
          }
        }
      }
    }
  }
`;
