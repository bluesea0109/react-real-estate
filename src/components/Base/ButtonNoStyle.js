import styled from 'styled-components';

const ButtonNoStyle = styled.button`
  background: transparent;
  padding: 0 1rem;
  border: none;
  font: inherit;
  font-size: 14px;
  font-weight: 600;
  color: white;
  cursor: pointer;
  border: none;
  &:focus {
    outline: none;
  }
`;

export default ButtonNoStyle;
