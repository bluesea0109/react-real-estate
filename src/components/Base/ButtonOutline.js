import styled from 'styled-components';

const ButtonOutline = styled.button`
  height: 36px;
  background: transparent;
  padding: 0 1rem;
  border: none;
  font: inherit;
  font-size: 14px;
  font-weight: 600;
  color: white;
  cursor: pointer;
  border: 2px solid white;
  border-radius: 36px;
  &:focus {
    outline: none;
  }
`;

export default ButtonOutline;
