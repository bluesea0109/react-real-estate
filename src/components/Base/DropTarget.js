import React, { useEffect, useRef } from 'react';

export default function DropTarget({ children, handleFileDrop, setDragging }) {
  const dropRef = useRef(null);
  const fileInput = useRef(null);
  let counter = 0;
  const handleDrag = e => {
    e.preventDefault();
    e.stopPropagation();
  };
  const handleDragIn = e => {
    e.preventDefault();
    e.stopPropagation();
    counter++;
    setDragging(true);
  };
  const handleDragOut = e => {
    e.preventDefault();
    e.stopPropagation();
    counter--;
    if (counter === 0) {
      setDragging(false);
    }
  };
  const handleDrop = e => {
    e.preventDefault();
    e.stopPropagation();
    setDragging(false);
    counter = 0;
    handleFileDrop(e.dataTransfer.files);
  };

  useEffect(() => {
    let target = dropRef?.current;
    target.addEventListener('dragenter', handleDragIn);
    target.addEventListener('dragleave', handleDragOut);
    target.addEventListener('dragover', handleDrag);
    target.addEventListener('drop', handleDrop);
    return () => {
      target.removeEventListener('dragenter', handleDragIn);
      target.removeEventListener('dragleave', handleDragOut);
      target.removeEventListener('dragover', handleDrag);
      target.removeEventListener('drop', handleDrop);
    };
    // eslint-disable-next-line
  }, [dropRef]);

  return (
    <>
      <div className="drop-target" ref={dropRef} onClick={() => fileInput.current.click()}>
        {children}
      </div>
      <input
        type="file"
        id="file"
        ref={fileInput}
        style={{ display: 'none' }}
        onChange={e => handleFileDrop(e.target.files)}
      />
    </>
  );
}
