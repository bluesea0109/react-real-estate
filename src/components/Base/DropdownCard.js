import React from 'react';
import styled from 'styled-components';
import { Icon } from '.';

const DropdownCard = ({ children, className, openCard, setOpenCard, iconName, title }) => {
  return (
    <div className={className}>
      <div
        className="card-title"
        onClick={() => (openCard === title ? setOpenCard('') : setOpenCard(title))}
      >
        <Icon name={iconName} />
        <h4>{title}</h4>
        <Icon className="arrow" name={openCard === title ? 'chevron up' : 'chevron right'}></Icon>
      </div>
      {openCard === title && <div className="content">{children}</div>}
    </div>
  );
};

const StyledDropdownCard = styled(DropdownCard)`
  box-shadow: 0 2px 8px 0 rgba(213, 213, 213, 0.5);
  border-radius: 7px;
  padding: 0.5rem 1rem;
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
  & > .card-title {
    font-size: 17px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    cursor: pointer;
    padding: 0.5rem 0;
    & > h4 {
      margin-left: 0.5rem;
      flex: 1;
    }
  }
  & > .content {
    margin-top: 0.5rem;
    transition: 1s;
  }
`;

export default StyledDropdownCard;
