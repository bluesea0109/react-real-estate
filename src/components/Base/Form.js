import { Form } from 'semantic-ui-react';
import styled from 'styled-components';

export default styled(Form)``;
