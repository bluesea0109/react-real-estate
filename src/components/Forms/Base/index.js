export { default as Debug } from './Debug';

export { default as Button } from './Button';
export { default as Form } from './Form';

export { default as Input } from './Input';
export { default as Checkbox } from './Checkbox';
export { default as Radio } from './Radio';
export { default as Dropdown } from './Dropdown';
export { default as TextArea } from './TextArea';

export { default as FileUpload } from './FileUpload';
