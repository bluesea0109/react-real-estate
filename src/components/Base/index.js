import Button from './Button';
import Container from './Container';
import Dimmer from './Dimmer';
import Divider from './Divider';
import Form from './Form';
import Grid from './Grid';
import Header from './Header';
import Icon from './Icon';
import Image from './Image';
import Input from './Input';
import Item from './Item';
import List from './List';
import Loader from './Loader';
import Menu from './Menu';
import Message from './Message';
import Modal from './Modal';
import Page from './Page';
import Pagination from './Pagination';
import Popup from './Popup';
import Responsive from './Responsive';
import Segment from './Segment';
import Select from './Select';
import Step from './Step';

export {
  Button,
  Container,
  Dimmer,
  Divider,
  Form,
  Grid,
  Header,
  Icon,
  Image,
  Input,
  Item,
  List,
  Loader,
  Menu,
  Message,
  Modal,
  Page,
  Pagination,
  Popup,
  Responsive,
  Segment,
  Select,
  Step,
};
