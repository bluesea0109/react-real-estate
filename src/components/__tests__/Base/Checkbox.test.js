import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Checkbox } from '../../Base';

it('renders correctly', () => {
  const tree = renderer.create(<Checkbox />);
  expect(tree).toMatchSnapshot();
});
