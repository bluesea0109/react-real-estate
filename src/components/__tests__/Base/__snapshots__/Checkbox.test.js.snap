// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c3 {
  fill: none;
  stroke: white;
  stroke-width: 2px;
}

.c2 {
  display: inline-block;
  width: 16px;
  height: 16px;
  background: white;
  border-radius: 3px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  border: 1px solid black;
}

<div
  className="c0"
>
  <input
    className="c1"
    type="checkbox"
  />
  <div
    className="c2"
  >
    <svg
      className="c3"
      viewBox="0 0 24 24"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </div>
</div>
`;
